src/server/protocol.ts

```diff
@@ -32,8 +32,10 @@ import {
 // These types/enums used to be defined in duplicate here and exported. They are re-exported to avoid breaking changes.
 export { ApplicableRefactorInfo, ClassificationType, CompletionsTriggerCharacter, CompletionTriggerKind, OrganizeImportsMode, RefactorTriggerReason, RenameInfoFailure, SemicolonPreference, SignatureHelpTriggerReason, SymbolDisplayPart, UserPreferences };
 
-type ChangeStringIndexSignature<T, NewStringIndexSignatureType> = { [K in keyof T]: string extends K ? NewStringIndexSignatureType : T[K]; };
-type ChangePropertyTypes<T, Substitutions extends { [K in keyof T]?: any; }> = {
+/** @internal */

```

Comment:
Internal isn't enough, as they are used within the public API.

Reply:
Well that’s a bummer

Reply:
I get why that can’t be dropped from declaration emit, but why do they need to be exported?

Reply:
Maybe I’ll work on patching the bundler to allow this

Reply:
The bundler is super dumb, and just walks exports and quite literally copies code over from the dts files. All it knows is when it's about to shadow another symbol of the same name. Then later, a test verifies that the code actually compiles, which only happens when the API is complete and consistent.

Theoretically we could mark some things as not being exported, but at the moment all of our namespaces are implicitly exporting everything, so we'd need to swap everything over to exports and then come up with some other tag that says to not actually export it, or something.

Or, completely redo the bundler to not be as dumb!

Reply:
I think our dts emitter works the same way isnt it.. it would want changePropertyTypes to be exported and if its marked internal the resulting d.ts would not have it and it would generate d.ts that has error..

Reply:
But i like this change though its not very readable..  So if we could write a script to generate these parts it might  be better.

Reply:
> it would want changePropertyTypes to be exported and if its marked internal the resulting d.ts would not have it and it would generate d.ts that has error..

I think that we'd actually just generate a namespace with export modifiers on everything, such that the helper type was not visible; see the dts emit here: [Playground Link](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAOwIYFtgGcxIMbDgMQgjgG8AoMuOGATzDwBEIBlCNGACwEsEBzAHgAqAPjgBeUnBAAuOILgBfANwUqAE2DYANkih4AZgFcE2GFwgI4alm2CceAkQAoZcgJSymrdtz5DhKpRwoJCwcNgW6PD6ROJWNj4OTgCMAEwAzAAsbioKQA)

The difference is that the bundler only walks the exported symbols only and assumes that the API is complete, not ever going through to find things it missed.